src/EventStore.Common/Configuration/ConfigurationBuilderExtensions.cs

```diff
@@ -28,5 +33,59 @@ public static class ConfigurationBuilderExtensions {
 
 			return builder;
 		}
+
+		// Allows configuration to be mounted inside a specified section
+		public static IConfigurationBuilder AddSection(
+			this IConfigurationBuilder self,
+			string sectionName,
+			Action<IConfigurationBuilder> configure) =>
+			
+			self.Add(new SectionSource(sectionName, configure));
+
+		public static IConfigurationBuilder AddConfigFile(
+			this IConfigurationBuilder builder,
+			string configFilePath,
+			bool optional = false,
+			bool reloadOnChange = false) {
+
+			if (!Locations.TryLocateConfigFile(configFilePath, out var directory, out var fileName)) {
```

Comment:
The function hides this custom behaviour that tries to locate the files in yet another path.
We should consider passing the path to the function upfront.